lib/util/git/index.ts

```diff
@@ -551,6 +551,30 @@ export async function isBranchModified(branchName: string): Promise<boolean> {
   return true;
 }
 
+export async function isBranchConflicted(
+  baseBranch: string,
+  branchName: string
+): Promise<boolean> {
+  if (!branchExists(branchName)) {
+    logger.debug({ branchName }, 'Conflict detection: branch does not exist');
+    return false;
+  }
+  try {
+    await syncGit();
+    await git.reset(ResetMode.HARD);
+    await git.checkout(branchName);
+    await git.merge(['--no-commit', '--no-ff', baseBranch]);
+    await git.reset(ResetMode.HARD);
+    await git.checkout(baseBranch);
+  } catch (err) {
+    if (err?.git?.conflicts?.length) {
+      return true;
+    }
+    logger.debug({ branchName, err }, 'Conflict detection: unknown error');
```

Comment:
All other errors as no conflict?

Reply:
We may return `null` for errors, but consuming part won't treat it differently. Alternatively, we may treat any error as `this brach is conflicted`, but most probably it's incorrect too 🤔